### ISSUE TITLE
[FEATURE] Marriage

### ISSUE THREAD

**Issue by beanbeanjuice** [beanbeanjuice/cafeBot#460]
**__Describe the Feature__**
Add the ability to "marry" other Discord users.

# Checklist:

- [x] I have checked for similar issues.
- [x] This is a feature request, not a bug report or improvement request.

**Comment by beanbeanjuice** [comment]
Need beanbeanjuice/cafe-api#9 to be completed first.

**Comment by beanbeanjuice** [comment]
"Divorce" should be possible without the partner's consent, as it would cause issues forcing people to have a consensual "divorce," however marriage should be 100% consensual.

This "consent" handshake can be done by setting a timer, the user must be mentioned, but the mentioned user will be sent a DM saying "so and so wants to marry you." They can do `/marry accept` or `/marry deny`.

FURTHERMORE, users should not be sent a DM if they are already married, and users can opt-out of receiving marriage requests, as that can be very obnoxious.

To summarise:
- [ ] One-Party Divorce
- [ ] Two-Party Marriage
- [ ] Handshake Timer System (LOCAL)
- [ ] Marriage Opt-Out
- [ ] No Marriage DM When Already Married
